src/__tests__/resources/assets.integration.spec.ts

```diff
@@ -35,11 +37,82 @@ describe('Asset integration test', () => {
   });
 
   test('create empty asset array', async () => {
-    await expect(
-      client.assets.create([])
-    ).rejects.toThrowErrorMatchingInlineSnapshot(
-      `"createAssets.arg0.items: size must be between 1 and 1000 | code: 400"`
+    const err = new CogniteMultiError({
+      succeded: [],
+      failed: [[]],
+      errors: [
+        new CogniteError(
+          'createAssets.arg0.items: size must be between 1 and 1000',
+          400
+        ),
+      ],
+      responses: [],
+    });
+    await client.assets.create([]).catch(e => {
+      expect(e.missing).toEqual(err.missing);
+      expect(e.failed).toEqual(err.failed);
+      expect(e.succeded).toEqual(err.succeded);
+      expect(e.status).toEqual(err.status);
+      expect(e.errors[0]).toEqual(err.errors[0]);
+      expect(e.responses).toEqual(err.responses);
+    });
+  });
+
+  test('create 1001 assets (1 corrupted): error handling', async () => {
+    const newRootAsset = {
+      ...rootAsset,
+      externalId: 'test-root' + randomInt(),
+    };
+    const corruptedChild = {
+      ...childAsset,
+      extraProp: 'corrupted',
+      parentExternalId: newRootAsset.externalId,
+    };
+    const newChildAsset = {
+      ...childAsset,
+      parentExternalId: newRootAsset.externalId,
+    };
+    const childArr = new Array(999).fill(newChildAsset);
+    try {
+      expect(
+        await client.assets.create([newRootAsset, ...childArr, corruptedChild])
+      ).rejects.toBeTruthy();
+    } catch (e) {
+      expect(e).toBeInstanceOf(CogniteMultiError);
```

Comment:
Will this run? Have you checked that `.rejects` in jest doesn't catch the exception?

Reply:
Yes, but I'll add
I will add `expect.assertions(8);`